CIP-0119? | Governance Metadata - DReps (#788)
cardano-foundation/CIPs: cip-drep-metadata/README.md

---
CIP: ?
Title: DRep Metadata

> Category: ?

Reviewer: ```suggestion
Category: Metadata
```
CIP meeting today confirmed category to match CIP-0100 and pending CIP-0108.

Status: Proposed
Authors:
    - Thomas Upfield <thomas.upfield@iohk.io>
Implementors: []
Discussions:
    - https://github.com/cardano-foundation/CIPs/pull/?
Created: 2024-02-07
License: CC-BY-4.0
---

<!-- Existing categories:

- Meta     | For meta-CIPs which typically serves another category or group of categories.
- Wallets  | For standardisation across wallets (hardware, full-node or light).
- Tokens   | About tokens (fungible or non-fungible) and minting policies in general.
- Metadata | For proposals around metadata (on-chain or off-chain).
- Tools    | A broad category for ecosystem tools not falling into any other category.
- Plutus   | Changes or additions to Plutus
- Ledger   | For proposals regarding the Cardano ledger (including Reward Sharing Schemes)
- Catalyst | For proposals affecting Project Catalyst / the Jörmungandr project

> -->

Reviewer: Please remove all this comment scaffolding from the CIP template.

Reviewer: Looks all gone now (in a later commit not linked back here).

Author: done. thx.

> ## Acknowledgements

Reviewer: We're standardising CIPs and CPSs which have more than one round of contributors and/or more than a modest size of text to link to the end... can you please follow the example of [CIP-1694](https://github.com/cardano-foundation/CIPs/blob/master/CIP-1694/README.md#abstract) to move this section to the end and (at your discretion, and following a general recommendation) to make a text comment at the end of the Abstract to link to it?

Reviewer: I see the section is moved (in a later commit not linked back here); and without the link which is optional... so resolving this.

Author: done.

There have been 3 lively public workshops on this subject, and I would like to thank the following people

### Workshop 1 - 07/02/2024

> Abhik Nag, 

Reviewer: ```suggestion
Abhik Nag
```
... etc. - None of our other documents have anything other than a newline / carriage return between names (no comma or trailing space).

- Adam Rusch,
- Aleksandar Djuricic (Aleks),
- Christopher Hockaday,
- Digital Fortress (Rick McCracken),
- Eduardo Silka,
- HD5000,
- Igor Velickovic,
- Input Endorsers,
- Konstantinos Dermentzis,
- Leonardo Silka,
- Michael Madoff,
- Michał Szałowski,
- Mike Hornan,
- Peter Wolcott,
- Ryan,
- Ryan (Cerkoryn),
- Ryan Williams,
- Steve Lockhart, 

### Austtralia and APAC Workshop  - 05/03/2024
- Andreas,
- Linh P,
- Mark Byers,
- Mike Hornan,
- Pedro Lucas,
- Phil Lewis,

## Abstract
The Conway ledger era ushers in on-chain governance for Cardano via [CIP-1694 | A First Step Towards On-Chain Decentralized Governance](https://github.com/cardano-foundation/CIPs/blob/master/CIP-1694/README.md), with the addition of many new on-chain governance artefacts. Some of these artefacts support the linking off-chain metadata, as a way to provide context.

The [CIP-100 | Governance Metadata](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0100) standard provides a base framework for how all off-chain governance metadata should be formed and handled. But this is intentionally limited in scope, so that it can be expanded upon by more specific subsequent CIPs.

> This proposal aims to provide a specification for off-chain metadata vocabulary that can be used to give context to CIP-100 for DRep registration, updates and retirement. Without a sufficiently detailed standard for DRep metadata we introduce the possibility to undermine the ability of DReps to explain their motivations and therefore people looking for someone to represent them with the best possible information available to make that choice. Furthermore a lack of such standards risks preventing interoperability between tools, to the detriment of user experiences.

Author: Yea I removed it from another section, but must've missed this part. Thanks for catching this.

## Motivation: why is this CIP necessary?
Blockchains are poor choices to act as content databases. This is why governance metadata anchors were chosen to provide a way to attach long form metadata content to on-chain events. By only supplying an On-Chain hash of the off-chain we ensure correctness of data whilst minimising the amount of data posted to the chain.

> #### For people wishing to delegate their voting power

Author: verbosity is my passion

> When observing from the chain level, tooling can only see the content and history of DRep registration, update and retirement certificates and any associated anchors. These on-chain components do not give any context to the motivation of a DRep, even though this information would likely be the desired context for people who might delegate their voting power. By providing rich contextual metadata we enable people choosing a DRep to delegate their voting power to, to make well informed decisions.

Author: I should really have done ctrl+f "retirement". Thanks again for catching this.

Participants at the workshop held to debut the first draft of this CIP strongly advocated in favour of a CIP that would include features that would assist a DRep comprised of a team of individuals properly describe their endeavour to delegating Ada Holders and anyone else reading the metadata associated with their registration.

This CIP has not included these features, the decision not to include these features was made in order to simplify this CIP so that it became suitable for the minimum viable level of tooling, with the expectation that further CIPs will build on it. 

## Specification
CIP-1694 specifies that metadata anchors are optional for DRep registration and updates. This CIP covers metadata for the aforementioned transaction types, but it does not cover metadata for voting transactions. 

### Markdown Text Styling
Unlike [CIP-108](https://github.com/Ryun1/CIPs/blob/governance-metadata-actions/CIP-0108/README.md#markdown-text-styling) this standard seeks to simplify the job of tooling providers by not supporting text formatting such as markdown. 

> ### Integrity

Author: I deleted it

Metadata is said to have integrity when the hash of this metadata in the metadata anchor on-chain is the same as a hash made of the metadata now (given that the same algorithm is used). Tooling SHOULD check for metadata integrity, with the hashing algorithm determined by rules outlined in [CIP-100](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0100). If tooling detects that the metadata does not have integrity it MUST NOT display the metadata, and it MUST display a warning describing why it is not showing the metadata.

> ### Formatting

Author: I deleted it

If DRep Metadata is not formatted in the way described by this CIP then tooling SHOULD either:
1. Deal with it as described in CIP-100 (by rendering the raw unformatted document)
2. Not display the any of the metadata
In either case the tooling SHOULD provide a warning that this has happened and link to the url described in the metadata anchor of the DRep registration.

### Missing
If the DRep metadata cannot be found at the address specified in the anchor then the tooling SHOULD provide a warning. There is no need to link to the url in the metadata anchor. 

### Witnesses
DRep Metadata will not follow the CIP-100 specification related to signing the metadata, instead they are attached to a DRep registration. The need for an `authors` field will also be discarded in favour of including usernames inside of the `body` field. 

### Extended Vocabulary

> Like CIP-108, this CIP also extends the potential vocabulary of CIP-100's `body` property. For all of these properties tooling providers MUST be aware that they are responsible for what they display to their users and that these fields could be used for illegal, unsavoury, or innapropriate purposes or language. Therefore such tools SHOULD have a terms of service which they enforce to moderate what they show to their users. The following are a list of properties tooling should expect to encounter:

Author: this would be defined by their TOS, but I was thinking of people using the metadata as a vector to be insulting or defamatory to other people or community members.

> Like CIP-108, this CIP also extends the potential vocabulary of CIP-100's `body` property. For all of these properties tooling providers MUST be aware that they are responsible for what they display to their users and that these fields could be used for illegal, unsavoury, or innapropriate purposes or language. Therefore such tools SHOULD have a terms of service which they enforce to moderate what they show to their users. The following are a list of properties tooling should expect to encounter:

Reviewer: I would be very worried about encoding centralised moderation or censorship into the CIP itself.  I believe the place for these things is in the tool documentation and, as you say, the user terms of service for those governance tools.  It is not part of the metadata framework itself to mandate censorship, and in fact that framework _itself_ will work fine without that censorship.  If there are reasons why such censorship is beneficial, or considered practically inevitable, then they should be documented in the practical materials for developing and using the tools, not the CIP: otherwise we invite a subjectivity into the CIP process that has long been avoided.

> Like CIP-108, this CIP also extends the potential vocabulary of CIP-100's `body` property. For all of these properties tooling providers MUST be aware that they are responsible for what they display to their users and that these fields could be used for illegal, unsavoury, or innapropriate purposes or language. Therefore such tools SHOULD have a terms of service which they enforce to moderate what they show to their users. The following are a list of properties tooling should expect to encounter:

Author: @rphair ok.

> Like CIP-108, this CIP also extends the potential vocabulary of CIP-100's `body` property. For all of these properties tooling providers MUST be aware that they are responsible for what they display to their users and that these fields could be used for illegal, unsavoury, or innapropriate purposes or language. Therefore such tools SHOULD have a terms of service which they enforce to moderate what they show to their users. The following are a list of properties tooling should expect to encounter:

Reviewer: @Thomas-Upfield thanks for your attention but there are a couple remaining issues with the relevant change in https://github.com/cardano-foundation/CIPs/pull/788/commits/c0fb5b6d6135c88c40d516d11fdabb95f1e71750:
- A short trailing paragraph has been, I assume inadvertently, included in the **Note**: introducing the list of properties: there would need to be a paragraph break before the final sentence.
- The issue is not resolved by changing the typography from a paragraph to a "Note" since it still makes the insupportable claims that the CIP can "enforce" censoring TOS's on tooling providers and also in turn mandate own "enforcement" of those terms on users.

There is a similar problem with the "intent" of the `username` field avoid "crass language"... no less a problem without an equally undignified inventory of which language is unacceptable (which itself would be "crass").

I suspect there is a bureaucratic mandate behind these repeated interjections of moderation (censorship) into what should be a straightforward technical document, so in my experience there would need to be a floodlight of visibility on this to decide how to move ahead as a community.

We have a light CIP agenda coming up so, given that this document is still being refined, I've added this one item to our agenda & would hope you can be there to explain the origin of this apparent moderation requirement and hopefully get some consensus about whether & how it might have a place in this (or any) CIP.  cc @Ryun1 @Crypto2099

`Payment Address`
Dreps may want to recieve tokens for a variety of reasons such as:
- donations
- expenses
- any incentive program

Therefore there will be a `payment_address` field in the metadata where such payments could be sent. This makes such an address public and payments to DReps transparent. 

`username`
- Compulsory
- A very short freeform text field. Limited to 80 characters.
- This SHOULD NOT support markdown text styling.
- Authors MUST use this field for their profile name/ username.
- Authors SHOULD attempt to make this field unique.
- Authours SHOULD avoid crass language.

> The compulsory nature of this field was controversial because the `username`s cannot be made unique and therefore are open to abuse (by e.g. copycats). However this is not a reason to not include a `username` it a reason for people reviewing governance actions to properly check the whole profile of a DRep before delegating to them. A `username` MUST be included because it is a human readable identifier, it is the property that people reviewing DReps will most likely identify that DRep by even in the presence of copycats.

Author: done.

`picture`
- Optional 
- A base 64 encoded profile picture
- Moderation of this image must be handled on the client side to comply with their TOS
- This SHOULD be treated as the profile picture of the individual

`objectives`
- Optional
- A freeform text field with a maximum of 1000 characters
- A short description of what the person believes nd what they want to achieve as a DRep

`motivation`
- Optional
- A freeform text field with a maximum of 1000 characters
- A short description of why they want to be a DRep, what personal and professional experiences have they had that have driven them to 

`qualifications`
- Optional
- A freeform text field with a maximum of 1000 characters 
- A space to list the qualifications that the subject of this metadata has that are relevant to being a DRep

> `identity`

Reviewer 2: why not reuse the author property from CIP-100?

> `identity`

Reviewer 2: ```suggestion
#### `identity`
```

- Compulsory
- A link to a social media profile where the person must reference their DRep ID in their profile.
- This will be used by people reviewing this DRep to prove and verify that the person described in the metadata is the same as the person who set up the social media profile.
- Tooling providers SHOULD warn people that none of the information is verified by the tool and they should DYOR
- Tooling providers making metadata SHOULD provide some information about why this is important.

> PKI cryptographic solutions were also considered in the place of the current solution however they were generally considered too complex to implement for minimum viable tooling. There were also other issues with cryptographic forms of identification for MVPs:

Reviewer 2: Move to `Rationale`

1. solutions that involve a public/private key setup still require people verifying the identity of a DRep to know what the authentic pulic key is.
2. specifying the use of a verification service such as [Keybase](https://keybase.io/) would lead to centralisation and therefore reliance on a 3rd party for DReps.

> `links`

Reviewer 2: why not reuse the References or externalUpdates from CIP100?

Author: lets discuss this.

Author: Ok, so the decision was that References should be used for all external links and that we can extend CIP-100 to add a @type called "identity" and possibly one called "link".

Author: @Ryun1 I have made this [change](https://github.com/Thomas-Upfield/CIPs/blob/drep-metadata/cip-drep-metadata/README.md#references)

> `links`

Reviewer 2: ```suggestion
#### `links`
```

- Optional
- A list of urls to the social media/ websites associated with the DRep
- It is up to the client to check where these links go
- Warning about linking to external links

> `donotlist`

Reviewer 2: ```suggestion
#### `donotlist`
```

- Optional
- If not included then the value is assumed to be false
- A boolean expression
- A true value means that the DRep does not want to show up in tooling that displays DReps. 
-- I.e. they do not want to appear in GovTool’s DRep Explorer feature

### Application

> DRep metadata must include all compulsory fields to be considered CIP-???? compliant. As this is an extension to CIP-100, all CIP-100 fields can be included within CIP-???? compliant metadata.

Reviewer 2: > all CIP-100 fields can be included within CIP-???? compliant metadata.

This confirmed?

Author: You are right to question this. The authors field has been explicitly rejected by this CIP.

Reviewer 2: > The authors field has been explicitly rejected by this CIP.

Or you could include it (makes it easier for tooling to comply)
but suggest the field is left blank?

Author: ok fixed. Thanks for the advice. https://github.com/Thomas-Upfield/CIPs/blob/drep-metadata/cip-drep-metadata/README.md#witnesses

### Test Vector
See test-vector.md for examples.

### Versioning
This proposal should not be versioned, to update this standard a new CIP should be proposed. Although through the JSON-LD mechanism further CIPs can add to the common governance metadata vocabulary,

## Rationale: how does this CIP achieve its goals?
We intentionally have kept this proposal brief and uncomplicated. This was to reduce the time to develop and deploy this standard. This way we enable tooling which depends on this standard to start development. The fields which have been chosen for this standard are heavily inspired by those that we are seeking to introduce for GovTool. We did this because GovTool will likely be the first technical implementation of this standard. 

> ## Open Questions

Reviewer: ```suggestion
### Open Questions
```
This is only a major section for CPSs so I'd suggest demoting it to a sub-section of _Rationale_ since these questions all appear related to the approach you've taken or are considering taking.

2. ~~Do we need to replace the `bio` field with a more structured set of fields~~
3. ~~Can we include and verify an ADA handle to uniquely identify a DRep~~
4. ~~What do we do about lack of metadata integrity~~ <-- not show the metadata and make it clear that the #metadata =/ metadata#
5. ~~Should we split this CIP up into separate transactions or also add the vote transaction metadata~~ <-- the scope is fine
6. ~~Compulsory vs optional for all fields~~
7. ~~types of DRep (script? representing an organisation? want delegations?)~~
8. ~~How can we verify the information that we are displaying(?)~~ 
9. ~~Tooling providers displaying data SHOULD warn people that none of the information is verified and they should DYOR~~
10. ~~Tooling providers making metadata SHOULD provide some information about best practices such as putting DRep ID in twitter bio.~~
11. ~~What do we care about when someone is retiring?~~ <-- just the reason why
12. ~~When someone updates do we want a 1 line summary of what has changed?~~
13. ~~Tooling to inform people of recent changes?~~
14. ~~incentives address, where DRep incentives are paid ???~~
15. ~~PKI Public Key, Fingerprint, or DID?~~ 

> ## Path to Active

Reviewer: Looking forward to seeing this section filled out in good time. 🤓

---
CIP: ?
Title: Governance metadata - DReps
Category: Metadata
Status: Proposed
Authors:
    - Thomas Upfield <thomas.upfield@iohk.io>
Implementors: []
Discussions:
    - https://github.com/cardano-foundation/CIPs/pull/788
Created: 2024-02-07
License: CC-BY-4.0
---

## Abstract

> The Conway ledger era ushers in on-chain governance for Cardano via [CIP-1694 | A First Step Towards On-Chain Decentralized Governance](https://github.com/cardano-foundation/CIPs/blob/master/CIP-1694/README.md), with the addition of many new on-chain governance artefacts. Some of these artefacts support the linking off-chain metadata, as a way to provide context.

Reviewer 3: ```suggestion
The Conway ledger era ushers in on-chain governance for Cardano via [CIP-1694 | A First Step Towards On-Chain Decentralized Governance](https://github.com/cardano-foundation/CIPs/blob/master/CIP-1694/README.md), with the addition of many new on-chain governance artefacts. Some of these artefacts support linking to off-chain metadata as a way to provide context.
```

#### Thank you
Thank you to [everyone who participated in the CIP workshops](#Acknowledgements)

## Motivation: why is this CIP necessary?
CIP-1694 has set forth a compelling vision of a blockchain controlled by its community, and in doing so has thrown down the gauntlet to tooling providers to build apps and other kinds of tooling that will allow their users easy enjoyment of the governance features currently being built into the fabric of the Cardano blockchain. As I write this developers are feverishly hammering away at their keyboards like the blacksmiths of old to bring us the first iteration of these tools in a minimum viable format. 

> The motivation for this CIP therefore is to provide these toolmakers with a simple and easy to accomodate standard which once adopted will allow them to read and display the metadata created by anyone who follows this standard when creating their DRep registration or update metadata. Tooling designed for DReps so that they can easily create metadata will also be made possible, because toolmakers will not need to all individually innovate over the contents or structure of the metadata that their tool creates.  

Reviewer 3: ```suggestion
The motivation for this CIP therefore is to provide these toolmakers with a simple and easy to accommodate standard which, once adopted, will allow them to read and display the metadata created by anyone who follows this standard when creating their DRep registration or update metadata. Tooling designed for DReps so that they can easily create metadata will also be made possible, because toolmakers will not need to individually innovate over the contents or structure of the metadata that their tool creates.  
```

#### A Note on Teams
CIP-1694 allows for DReps to be registered using a native or Plutus script credential, this implies that individuals could organise to form a team that would have a broad range of experitise, and would perhaps therefore be more attractive to some delegating Ada Holders. 

Participants at the workshop held to debut the first draft of this CIP strongly advocated in favour of a CIP that would include features that would assist a DRep comprised of a team of individuals properly describe their endeavour to delegating Ada Holders and anyone else reading the metadata associated with their registration.

This CIP has not included these features, the decision not to include these features was made in order to simplify this CIP so that it became suitable for the minimum viable level of tooling, with the expectation that further CIPs will build on it. 

### Formatting
If DRep Metadata is not formatted in the way described by this CIP then tooling SHOULD either:
1. Deal with it as described in CIP-100 (by rendering the raw unformatted document)

> 2. Not display the any of the metadata

Reviewer 4: Point 2. should read "2. Not display any of the metadata"

### Extended Vocabulary
Like CIP-108, this CIP also extends the potential vocabulary of CIP-100's `body` property. For all of these properties tooling providers MUST be aware that they are responsible for what they display to their users and that these fields could be used for illegal, unsavoury, or innapropriate purposes or language. Therefore such tools SHOULD have a terms of service which they enforce to moderate what they show to their users. The following are a list of properties tooling should expect to encounter:

#### `paymentAddress`
Dreps may want to recieve tokens for a variety of reasons such as:
- donations
- expenses
- any incentive program

Therefore there will be a `payment_address` field in the metadata where such payments could be sent. This makes such an address public and payments to DReps transparent. 

#### `username`
- Compulsory
- A very short freeform text field. Limited to 80 characters.

> - This SHOULD NOT support markdown text styling.

Reviewer 3: ```suggestion
- This MUST NOT support markdown text styling.
```
I would change this to a "MUST NOT" because making it "SHOULD NOT" means that someone inevitably will use markdown or emojis or other "rich elements" that then force all downstream tooling to support them (see the whole SPO Ticker Debacle).

Reviewer: yes, we MUST use more emphatic language when there is something that MUST be done (even though we also "should" follow this rule) 🧐

- Authors MUST use this field for their profile name/ username.
- Authors SHOULD attempt to make this field unique.

> - Authours SHOULD avoid crass language.

Reviewer 3: ```suggestion
- Authors SHOULD avoid crass language.
```

Author: I have resolved this, thanks.

The compulsory nature of this field was controversial because the `username`s cannot be made unique and therefore are open to abuse (by e.g. copycats). However this is not a reason to not include a `username` it a reason for people reviewing governance actions to properly check the whole profile of a DRep before delegating to them. A `username` MUST be included because it is a human readable identifier, it is the property that people reviewing DReps will most likely identify that DRep by even in the presence of copycats.

#### `picture`
- Optional 
- A base 64 encoded profile picture

> - Moderation of this image must be handled on the client side to comply with their TOS

Reviewer 3: As mentioned elsewhere, is it necessary to specify when and where clients should moderate the contents of the metadata rather than just describing the metadata format itself?

Author: I have resolved this now. Thanks for your input.

Reviewer: @Crypto2099 @Thomas-Upfield it's disputable whether this is resolved (see https://github.com/cardano-foundation/CIPs/pull/788#discussion_r1548122679) so please let's agree upon a stance regarding moderation at the next CIP meeting (in a few minutes) if possible.

- This SHOULD be treated as the profile picture of the individual

#### `objectives`
- Optional
- A freeform text field with a maximum of 1000 characters

> - A short description of what the person believes nd what they want to achieve as a DRep

Reviewer 4: small typo "believes **and** what..."

Author: thanks for pointing this out, I have resolved it

#### `motivation`
- Optional
- A freeform text field with a maximum of 1000 characters
- A short description of why they want to be a DRep, what personal and professional experiences have they had that have driven them to 

#### `qualifications`
- Optional
- A freeform text field with a maximum of 1000 characters 

> - A space to list the qualifications that the subject of this metadata has that are relevant to being a DRep

Author: > rationale about having two fields
Compared to a single field such as "bio" multiple fields elicit better responses. There fore this is not the same as a single field.
> The beauty of using JSON-LD as a format is that we can define and scope that qualifications is a list of one or more qualification objects...
Yes but we don't yet know how people will want to format this, therefore for MVP I have left it as freeform, knowing that this standard will need to be built on.

> - A space to list the qualifications that the subject of this metadata has that are relevant to being a DRep

Author: I will add my rationale for this to the rationale section

#### `references`
- This CIP extends the `references` property from [CIP-100](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0100#high-level-description)
- `references` contain the following sub-properties `type`, `label`, and `uri`
- This CIP adds two `@type` identifiers "Identity" and "Link"
- The creator of the metadata MUST add a `label`, this `label` SHOULD describe the source of the url, e.g. if it is a link to the DRep's X account then the `label` should say "X". If it is a personal website the `label` should say "Personal Website" rather than domain_name.com.
- The `label` of each link SHOULD NOT be left blank, but if it is then tooling displaying links SHOULD still allow users to access the URL.
- Each link MUST have exactly one `uri` (as specified in CIP-100) which MUST not be blank. If it is blank then tooling providers displaying the Link SHOULD NOT display any of the details of this link.

> - `uri` MUST be the full address to the souce of the intended data and MUST NOT be a shortened url i.e. one that has been created using a service such as bitly. This is to ensure that tooling can easily scan for malicious links and users of tooling who are reading the `uri` value can understand where the link goes.

Reviewer 3: How can you enforce that this is not a shortened URL?

Author: I have removed this

- Tools that have a user interface that allows users to navigate to the location in the `uri` of a link, SHOULD inform their user of the full url that they will be visiting and SHOULD also warn their user that they will be visiting a different website and ask them whether they want to proceed.

> - Tooling providers SHOULD warn people that none of the information contained in external is verified by the tool and they should DYOR

Reviewer: @Thomas-Upfield @Crypto2099 this is related to the still-unresolved question in https://github.com/cardano-foundation/CIPs/pull/788#discussion_r1533301310.

> - Tooling providers SHOULD warn people that none of the information contained in external is verified by the tool and they should DYOR

Author: I have removed this, but will add a note to tooling providers as a reminder to them rather than a prescription about whether they MUST or should do something.

##### `type`: Link
- Optional
- It is expected that these links will be the addresses of the social media/ websites associated with the DRep in order to give a person reviewing this information a fulsome overview of the DRep's online presence.

##### `type`: Identity
- The `uri` of a reference with the `type` "Identity" is a way for DReps to prove that they are who they say they are

> - Each piece of metadata referenced in the anchor of a DRep registristration or update transaction MUST contain at least one reference with the `type` "Identity". Else tooling SHOULD ignore the metadata.

Author: For example there are 5 DReps all with the username "Charles Hoskinson" this is a solution for how a potential delegator can discern which, if any, is the actual CH.

> - Each piece of metadata referenced in the anchor of a DRep registristration or update transaction MUST contain at least one reference with the `type` "Identity". Else tooling SHOULD ignore the metadata.

Author: All DReps should include some form of verification.

---

> CIP: ?

Reviewer: ```suggestion
CIP: 119
```

#### A Note on Teams
CIP-1694 allows for DReps to be registered using a native or Plutus script credential, this implies that individuals could organise to form a team that would have a broad range of experitise, and would perhaps therefore be more attractive to some delegating Ada Holders. 

Participants at the workshop held to debut the first draft of this CIP strongly advocated in favour of a CIP that would include features that would assist a DRep comprised of a team of individuals properly describe their endeavour to delegating Ada Holders and anyone else reading the metadata associated with their registration.

This CIP has not included these features, the decision not to include these features was made in order to simplify this CIP so that it became suitable for the minimum viable level of tooling, with the expectation that further CIPs will build on it. 

### Formatting
If DRep Metadata is not formatted in the way described by this CIP then tooling SHOULD either:
1. Deal with it as described in CIP-100 (by rendering the raw unformatted document)
2. Not display any of the metadata

> In either case the tooling SHOULD provide a warning that this has happened and link to the url described in the metadata anchor of the DRep registration.

Reviewer: @Thomas-Upfield I can see this passage doesn't appear anymore (so resolving).  From the clip above (out of context) I can't think of any reason why it that process would be vital & therefore should be OK to delete the suggestion.

### Missing
If the DRep metadata cannot be found at the address specified in the anchor then the tooling SHOULD provide a warning. There is no need to link to the url in the metadata anchor. 

### Witnesses
DRep Metadata will not follow the CIP-100 specification related to signing the metadata, the [authors property](https://github.com/cardano-foundation/CIPs/blob/master/CIP-0100/README.md#high-level-description) can be left blank without the need for tooling providers to warn their users that the author has not been validated. Instead the author can be derived from the DRep ID associated with the registration. The need for an `authors` field will also be discarded in favour of including usernames inside of the `body` field. For the avoidance of doubt this CIP recommends that the entire authors property be left blank, and that tooling ignore it. 

### Extended Vocabulary
Like CIP-108, this CIP also extends the potential vocabulary of CIP-100's `body` property. 

> >**Note** For all of these properties tooling providers will need to be aware that they are responsible for what they display to their users and that these fields could be used for illegal, unsavoury, or innapropriate purposes or language. Therefore they will be responsible for having a terms of service which they enforce to moderate what they show to their users.

Reviewer 2: ```suggestion
>**Note** For all of these properties tooling providers will need to be aware that they are responsible for what they display to their users and that these fields could be used for illegal, unsavoury, or inappropriate purposes or language.
```
remove part of enforcement, just leave it as a warning to implementors?
cc @Crypto2099 @rphair

Author: I think that this note should be changed to a reminder for tooling providers, to remind them that when they display information from DRep metadata that they will be displaying user generated content. And that for whatever reason, they may decide that they do not want to display certain user generated content to their other users. Therefore they may also need to introduce a TOS and some form of moderation to handle this.

Author: I say this because its something that I easily forget and I have noticed this forgetfulness in other people also.

Reviewer: thanks @Thomas-Upfield - there are three parts to this proposition, so I wanted to address all of them individually:
> a reminder for tooling providers, to remind them that when they display information from DRep metadata that they will be displaying user generated content.

👍 1) This is correct and sufficient: and also well-precented in the security field in considering information "tainted" whenever it is entered by a user.  Implementors, users, and potential regulators will be able to determine rights and responsibilities form this statement alone.

> And that for whatever reason, they may decide that they do not want to display certain user generated content to their other users.

👎 2) This is not necessary and leads immediately to the same red flags we pointed out at the meeting today: "condescension" I think we called it, and a suggestion of "nannying" that is considered obligatory according to one cultural bias but unacceptable in another.

If we suggest the censorship, it will be considered an _recommendation_ or at least an _endorsement_ of censorship, as well as an invitation to those potential regulators... so the "disclaimer" does not actually disclaim anything and will in fact be seen as an acceptance of liability that we wouldn't otherwise have.

> Therefore they may also need to introduce a TOS and some form of moderation to handle this.

👎 3) Also unnecessary for the same reason.  Making point #1 above is enough: to say that the tooling providers are posting user data to the blockchain.  They may "need" moderation and may not, and nobody can anticipate what their jurisdiction or legal precedent might be.  If "we" tell them what to do, "we" become _responsible_ for what they do: and the "potential regulators" are suddenly interested in who "we" are.

To everyone else, the presumption of a hypothetical censor _creates_ this censor: a centralised component in an otherwise decentralised process.  We need to say as little as possible about the moderation issue and simply tell tooling providers they are conveying / displaying / propagating user generated content.  Each tooling provider will have their own notion of liability and we cannot afford for this to be either our technical or our legal concern.

Reviewer: ... cross-referencing the related review points (and hoping further responses can continue in a single thread here, then resolved in those threads when resolved here):
https://github.com/cardano-foundation/CIPs/pull/788#discussion_r1533301310
https://github.com/cardano-foundation/CIPs/pull/788#discussion_r1548122679
https://github.com/cardano-foundation/CIPs/pull/788#discussion_r1546389557
https://github.com/cardano-foundation/CIPs/pull/788#discussion_r1546394333

The following are a list of properties tooling should expect to encounter:

#### `paymentAddress`

> Dreps may want to recieve tokens for a variety of reasons such as:

Reviewer 2: ```suggestion
DReps may want to recieve tokens for a variety of reasons such as:
```

> Dreps may want to recieve tokens for a variety of reasons such as:

Reviewer 2: ```suggestion
- Optional

DReps may want to receive tokens for a variety of reasons such as:
```

optional?

> - donations

Reviewer 2: ```suggestion
- Optional
- donations
```
optional?

#### `username`
- Compulsory
- A very short freeform text field. Limited to 80 characters.
- This MUST NOT support markdown text styling.
- It is intended that authors will use this field for their profile name/ username.
- It is intended that authors will attempt to make this field unique.
- It is intended that authors will avoid crass language.

#### `picture`
- Optional 
- A base 64 encoded profile picture
- This SHOULD be treated as the profile picture of the individual

#### `objectives`
- Optional
- A freeform text field with a maximum of 1000 characters
- A short description of what the person believes and what they want to achieve as a DRep

#### `motivations`
- Optional
- A freeform text field with a maximum of 1000 characters
- A short description of why they want to be a DRep, what personal and professional experiences have they had that have driven them to 

#### `qualifications`
- Optional
- A freeform text field with a maximum of 1000 characters 
- A space to list the qualifications that the subject of this metadata has that are relevant to being a DRep

> #### `references`

Reviewer 2: ```suggestion
#### `references`
- Optional
```

- This CIP extends the `references` property from [CIP-100](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0100#high-level-description)
- `references` contain the following sub-properties `type`, `label`, and `uri`
- This CIP adds two `@type` identifiers "Identity" and "Link"
- The creator of the metadata MUST add a `label`, this `label` SHOULD describe the source of the url, e.g. if it is a link to the DRep's X account then the `label` should say "X". If it is a personal website the `label` should say "Personal Website" rather than domain_name.com.
- The `label` of each link SHOULD NOT be left blank, but if it is then tooling displaying links SHOULD still allow users to access the URL.
- Each link MUST have exactly one `uri` (as specified in CIP-100) which MUST not be blank. If it is blank then tooling providers displaying the Link SHOULD NOT display any of the details of this link.

> ##### `type`: Link

Reviewer 2: what is the difference between this and uri?

- It is expected that the "Identity" of a DRep will be the addresses of their most active social media account twitter, linkedin etc. or personal website.
- The DRep must reference their DRep ID in a prominent place at the location that is specified in the `uri` property of that reference. This will be used by people reviewing this DRep to prove and verify that the person described in the metadata is the same as the person who set up the social media profile.
- Tooling providers making metadata SHOULD provide some information about why this is important.

PKI cryptographic solutions were also considered in the place of the current solution however they were generally considered too complex to implement for minimum viable tooling. There were also other issues with cryptographic forms of identification for MVPs:
1. solutions that involve a public/private key setup still require people verifying the identity of a DRep to know what their authentic public key is.
2. specifying the use of a verification service such as [Keybase](https://keybase.io/) would lead to centralisation and therefore reliance on a 3rd party for the identity validation of DReps.

#### `donotlist`

> - Optional

Reviewer 3: What do you do if they don't provide it? If compulsory shouldn't it be like:
```
publicListing: true
```
So that it is an explicit opt-in rather than a required opt-out: false?

- If not included then the value is assumed to be false
- A boolean expression
- A true value means that the DRep does not want to show up in tooling that displays DReps. 
-- I.e. they do not want to appear in GovTool’s DRep Explorer feature

### Application
DRep metadata must include all compulsory fields to be considered CIP-???? compliant. As this is an extension to CIP-100, all CIP-100 fields can be included within CIP-???? compliant metadata.

### Test Vector

> See test-vector.md for examples.

Reviewer 2: ```suggestion
See [test-vector.md](./test-vector.md) for examples.
```

### Versioning

> This proposal should not be versioned, to update this standard a new CIP should be proposed. Although through the JSON-LD mechanism further CIPs can add to the common governance metadata vocabulary,

Reviewer 2: ```suggestion
This proposal should not be versioned, to update this standard a new CIP should be proposed. Although through the JSON-LD mechanism further CIPs can add to the common governance metadata vocabulary.
```

## Rationale: how does this CIP achieve its goals?

> We intentionally have kept this proposal brief and uncomplicated. This was to reduce the time to develop and deploy this standard. This way we enable tooling which depends on this standard to start development. The fields which have been chosen for this standard are heavily inspired by those that we are seeking to introduce for GovTool. We did this because GovTool will likely be the first technical implementation of this standard. 

Reviewer 2: probably no need to describe GovTool, doesnt add to the Rationale

I believed that this CIP would provide a benefit to:

#### Potential delegators

> When observing from the chain level, tooling can only see the content and history of DRep registration and update certificates and any associated anchors. These on-chain components do not give any context to the motivation of a DRep, even though this information would likely be the desired context for people who might delegate their voting power. By providing rich contextual metadata we enable people choosing a DRep to delegate their voting power to make well informed decisions.

Reviewer 2: This is more `Motivation` than `Rationale`

Reviewer: dup https://github.com/cardano-foundation/CIPs/pull/788#discussion_r1549605293

> #### DReps

Reviewer 2: this is more `Motivation` than `Rationale`

> DReps will be able to use tools that create metadata in a standard format. This in turn will allow their metadata to be read by apps that will render their words on screen for potential delegating Ada Holders to enjoy, this may lead to greater levelso of delegation. 

Reviewer 2: ```suggestion
DReps will be able to use tools that create metadata in a standard format. This in turn will allow their metadata to be read by apps that will render their words on screen for potential delegating Ada Holders to enjoy, this may lead to greater levels of delegation. 
```
spelling

> #### All participants

Reviewer 2: this is more `Motivation` than `Rationale`

By standardising off-chain metadata formats for any tooling which creates and/or renders metadata referenced in DRep registration and update transactions we facilitate interoperability. This in turn promotes a rich user experience between tooling. This is good for all governance participants.

### Rationale for Insisting on a compulsory username

> The compulsory nature of this field was controversial because the `username`s cannot be made unique and therefore are open to abuse (by e.g. copycats). However this is not a reason to not include a `username` it a reason for people reviewing governance actions to properly check the whole profile of a DRep before delegating to them. A `username` MUST be included because it is a human readable identifier, it is the property that people reviewing DReps will most likely identify that DRep by even in the presence of copycats.

Reviewer 2: > a reason for people reviewing governance actions to properly check the whole profile of a DRep before

do you mean reviewing DRep profiles?
